test/unit/core/simd/view_to_simd_test.cpp

```diff
@@ -0,0 +1,267 @@
+                                    std::tuple<std::vector<dna4>, simd_type_t<uint64_t>>,
+                                    std::tuple<std::deque<dna4>, simd_type_t<int8_t>>,
+                                    std::tuple<std::deque<dna4>, simd_type_t<int16_t>>,
+                                    std::tuple<std::deque<dna4>, simd_type_t<int32_t>>,
+                                    std::tuple<std::deque<dna4>, simd_type_t<int64_t>>,
+                                    std::tuple<std::deque<dna4>, simd_type_t<uint8_t>>,
+                                    std::tuple<std::deque<dna4>, simd_type_t<uint16_t>>,
+                                    std::tuple<std::deque<dna4>, simd_type_t<uint32_t>>,
+                                    std::tuple<std::deque<dna4>, simd_type_t<uint64_t>>
+                                   >;
+TYPED_TEST_CASE(view_to_simd_fixture, test_types);
+
+TEST(view_to_simd, concept_check)
+{
+    using cmp_type = std::vector<dna4_vector>;
+    using test_type = detail::view_to_simd<all_view<cmp_type &>, simd_type_t<int8_t>>;
+
+    using iter_t = decltype(std::ranges::begin(std::declval<test_type &>()));
+    EXPECT_TRUE(std::InputIterator<iter_t>);
+
+    EXPECT_EQ(std::ranges::InputRange<cmp_type>, std::ranges::InputRange<test_type>);
+    EXPECT_NE(std::ranges::ForwardRange<cmp_type>, std::ranges::ForwardRange<test_type>);
+    EXPECT_NE(std::ranges::BidirectionalRange<cmp_type>, std::ranges::BidirectionalRange<test_type>);
+    EXPECT_NE(std::ranges::RandomAccessRange<cmp_type>, std::ranges::RandomAccessRange<test_type>);
+    EXPECT_NE(std::ranges::RandomAccessRange<cmp_type>, std::ranges::RandomAccessRange<test_type>);
+
+    EXPECT_EQ(std::ranges::Range<cmp_type>, std::ranges::Range<test_type>);
+    EXPECT_NE(std::ranges::View<cmp_type>, std::ranges::View<test_type>);
+    EXPECT_EQ(std::ranges::SizedRange<cmp_type>, std::ranges::SizedRange<test_type>);
+    EXPECT_NE(std::ranges::CommonRange<cmp_type>, std::ranges::CommonRange<test_type>);
+    EXPECT_NE(ConstIterableRange<cmp_type>, ConstIterableRange<test_type>);
+    EXPECT_NE((std::ranges::OutputRange<cmp_type, dna4_vector>), (std::ranges::OutputRange<test_type, dna4_vector>));
+}
+
+TEST(view_to_simd, iter_concept)
+{
+    using cmp_type = std::vector<dna4_vector>;
+    using test_type = detail::view_to_simd<all_view<cmp_type &>, simd_type_t<int8_t>>;
+    using iter_t = std::ranges::iterator_t<test_type>;
+    using sent_t = std::ranges::sentinel_t<test_type>;
+
+    EXPECT_TRUE(std::Iterator<iter_t>);
+    EXPECT_TRUE(std::InputIterator<iter_t>);
+    EXPECT_FALSE(std::ForwardIterator<iter_t>);
+    EXPECT_FALSE(std::BidirectionalIterator<iter_t>);
+    EXPECT_FALSE(std::RandomAccessIterator<iter_t>);
+    EXPECT_FALSE((std::OutputIterator<iter_t, decltype(*std::declval<iter_t &>())>));
+    EXPECT_TRUE((std::Sentinel<sent_t, iter_t>));
+}
+
+TYPED_TEST(view_to_simd_fixture, size)
```

Comment:
I make a TypedTest of view_to_simd using the fixture. I don't understand your question.

Reply:
Well I always thought that the naming scheme is: TYPED_TEST(test_name, what_will_be_tested)

That means, what is the name of your test? It is view_to_simd_fixture. So you test a fixture. That sounds weird. We sometimes use `blah_test` as name. I think it would be a better suffix.

Reply:
Well, yeah, but then everywhere else in the library they started to use fixture as name. Like iterator_fixture. But I gladly change that

Reply:
i changed it